src/typeahead/typeahead.js

```diff
@@ -251,6 +251,11 @@ angular.module('ui.bootstrap.typeahead', ['ui.bootstrap.position', 'ui.bootstrap
         $setModelValue(originalScope, model);
         modelCtrl.$setValidity('editable', true);
 
+        if (modelCtrl.$$lastCommittedViewValue === modelCtrl.$viewValue) {
+          // angular will check if model value changed to run validators, so we need to run it manually
+          modelCtrl.$$runValidators(undefined, modelCtrl.$modelValue, modelCtrl.$viewValue, angular.noop);
```

Comment:
These are undocumented API. Is there any way to this with the documented API (prefixed with a single `$`)?